llvm/include/llvm/CodeGen/StackMaps.h

```diff
@@ -196,11 +196,11 @@ class StackMaps {
     unsigned Size = 0;
     unsigned Reg = 0;
     int64_t Offset = 0;
-    unsigned ExtraReg = 0;
+    std::set<int64_t> Extras;
```

Comment:
Dumb question: are these `int64_t`s spread all over the 64 bit space or do they go from 0..n without gaps or ... ?

Reply:
If I understand right, positive numbers are dwarf register numbers, and negative numbers are stack offsets, so the values are sparse.

Reply:
all over the place. An extra can be a register `1, 2, 3, 15` or an offset `-10, -2400`.

Reply:
> If I understand right, positive numbers are dwarf register numbers, and negative numbers are stack offsets, so the values are sparse.

Oh! Where's that documented?

Reply:
Right here: [1ebceb2](https://github.com/ykjit/ykllvm/pull/204/commits/1ebceb2acd5009826f0651eae5bb7529aebbdd88) :stuck_out_tongue: